packages/preview/paddling-tongji-thesis/0.1.1/paddling-tongji-thesis/tongjithesis.typ

```diff
@@ -0,0 +1,161 @@
+        if it.numbering != none {
+          h(-2em)
+          numbering(it.numbering, ..counter(heading).at(loc))
+          h(1em)
+          it.body
+        } else {
+          it
+        }
+        v(1em)
+      } else if it.level == 3 {
+        set text(font: font-family.hei, size: font-size.at("5"), weight: "bold")
+        v(0.5em)
+        if it.numbering != none {
+          numbering(it.numbering, ..counter(heading).at(loc))
+          h(1em)
+          it.body
+        } else {
+          it
+        }
+        v(1em)
+      } else if it.level == 4 {
+        set text(font: font-family.hei, size: font-size.at("5"), weight: "bold")
+        v(-0.5em)
+        grid(columns: (2em, 1fr), [], it)
+        v(0.5em)
+      } else if it.level == 5 {
+        set text(font: font-family.hei, size: font-size.at("5"), weight: "bold")
+        v(-0.5em)
+        grid(columns: (2em, 1fr), [], it)
+        v(0.5em)
+      } else {
+        it
+      }
+    },
+  ) + empty-par()
+
+  show list: it => it + empty-par()
+  show enum: it => it + empty-par()
+  show figure: it => v(0.5em) + it + v(0.5em) + empty-par()
+  show figure: set block(breakable: true)
+  show table: it => it + empty-par()
+  show math.equation.where(block: true): it => it + empty-par()
+  show raw.where(block: true): it => it + empty-par()
+
+  show heading: i-figured.reset-counters.with(extra-kinds: ("algo",))
+  show figure: i-figured.show-figure.with(extra-prefixes: (algo: "algo:"))
+  show math.equation.where(block: true): i-figured.show-equation
+  show figure.where(kind: table): set figure.caption(position: top)
+
+  set page(
+    numbering: "I", header: {
+      set text(font: font-family.song, font-size.at("-4"))
+      grid(
+        columns: (0.5em, 1fr, auto, 0.5em), [], image("figures/tongji.svg", height: 1cm), block(height: 0.7cm, [#set align(right); 毕业设计（论文）]), [],
+      )
+      v(-0.5em)
+      line(length: 100%, stroke: 1.8pt)
+      draw-binding()
+    }, header-ascent: 20%, footer: locate(loc => {
+      set align(center)
+      set text(font: font-family.song, size: font-size.at("-4"))
+      numbering("I", counter(page).at(loc).first())
+    }),
+  )
+  counter(page).update(1)
+
+  make-abstract(
+    title: title, abstract: abstract, keywords: keywords, prompt: ("摘要", "关键词："),
+  )
+  pagebreak()
+
+  make-abstract(
+    title: title-english, abstract: abstract-english, keywords: keywords-english, prompt: ("ABSTRACT", "Key words: "), is-english: true,
+  )
+  pagebreak()
+
+  make-outline()
+  pagebreak()
+
+  set page(footer: locate(loc => {
+    line(stroke: 1.8pt, length: 100%)
+    set align(right)
+    set text(font: font-family.song, size: font-size.at("-4"))
+    v(-0.6em)
+    [
+      共#h(1em)
+      #counter(page).final(loc).at(0)#h(1em)
+      页#h(1em)
+      第#h(1em)
+      #counter(page).display()
+      #h(1em)页
+    ]
+  }))
+  counter(page).update(1)
+
+  doc
+}
+
+#let make-bib(bib_dir: "bib/note.bib", full: false) = {
```

Comment:
To be more coherent with the rest of the package, it would be nice to use kebab-case here and rename the argument to `bib-dir`.